src/player/PlayerEquipment.gd

```diff
@@ -31,9 +34,34 @@ func _ready() -> void:
 func get_inventory() -> Inventory:
 	return _inventory;
 
+
+func _on_inventory_inventory_event(item_stack, cursur_position, ev) -> void:
+	if ev is InputEventMouseButton && ev.pressed && ev.button_index == MOUSE_BUTTON_LEFT:
+		var hovered_rect: Rect2i;
+		
+		if item_stack == null:
+			hovered_rect = Rect2i(cursur_position, Vector2i.ONE);
+		else:
+			hovered_rect = Rect2i(item_stack.position, item_stack.get_rotated_size());
+		
+		var selection: InventoryRenderer.ItemSelection = _inventory_renderer.get_selection(ID_SELECTION)
+		if selection != null and selection.get_rect() == Rect2i(hovered_rect.position, hovered_rect.size):
+			_inventory_renderer.clear_selection(ID_SELECTION);
+		elif item_stack == null:
+			_inventory_renderer.clear_selection(ID_SELECTION);
+		else:
+			var color: Color = get_theme_color("selected_color", "InventoryRenderer");
+			var outline: Color = get_theme_color("selected_outline", "InventoryRenderer");
+			
+			_inventory_renderer.add_selection(ID_SELECTION, hovered_rect, color, outline);
+		
+		if selection != null and selection.get_rect() != InventoryRenderer.ItemSelection.NO_RECT:
```

Comment:
why is that second check there?

Reply:
and the first one too. if LMB was pressed, select the passed in item stack, or null if nothing was clicked on. it's just select(item_stack) behind that is LMB pressed if

Reply:
i don't see why we should couple the animations with selection logic